src/metabase/db/custom_migrations.clj

```diff
@@ -1266,3 +1267,7 @@
     (qs/delete-trigger scheduler (triggers/key "metabase.task.send-pulses.job"))
     (qs/delete-job scheduler (jobs/key "metabase.task.send-pulses.trigger"))
     (qs/shutdown scheduler)))
+
+(define-reversible-migration MigrateMetricsToV2
+  (metrics-v2/migrate-up)
+  (metrics-v2/migrate-down))
```

Comment:
This is a nitpick but destructive thread-unsafe stuff like this should get exclamation points at the end. I know, who cares, right? Well being consistent about the pattern encourages other people to be consistent about it too. And we do have the linter that checks for thread-unsafe destructive stuff based on the presence of `!` at the end in `^:parallel` tests and warns you not to use them. A call to `metrics-v2/migrate-down!` is almost certainly not something you would want in a `^:parallel` test, right? So if you give that a `!` you get the Kondo linting for free.

```suggestion
(define-reversible-migration MigrateMetricsToV2
  (metrics-v2/migrate-up!)
  (metrics-v2/migrate-down!))
```

Reply:
The other benefit of consistently using `!` is that it makes it clear which functions are the unsafe ones doing IO and which ones are just pure functions manipulating immutable data. For example, you could assume `replace-metric-refs!` is hitting the app DB but `replace-metric-refs` is not. This only works if we follow this pattern consistently